app/models/like.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class Like < ApplicationRecord
+  belongs_to :user
+  belongs_to :tweet
+
+  validates :user_id, uniqueness: { scope: :tweet_id, message: :already_liked }
+  validate :validate_user_cannot_like_own_tweet
+
+  private
+
+  def validate_user_cannot_like_own_tweet
+    return if tweet.blank? || user.blank?
+
+    errors.add(:base, :cannot_like_own_tweet) if user_id == tweet.user_id
```

Comment:
what do you think about this:
```suggestion
    return unless user_id.present? && user_id == tweet.user_id

    errors.add(:base, :cannot_like_own_tweet)
```